### PR TITLE
Command new

### PR DESCRIPTION
**Pull request by denCalderon**
The command new will create a directory structure with some templates for files as README, cmake, etc

### CODE REVIEW

**Review by hamishun — DISMISSED** [review]

File: kit/command_new.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    specs = Spec(component, instance_spec, "")
+    specs.to_toml_file(toml_path)
+
+
+def create_cmake_template(project_name: str, cmake_path: Path):
+    """Create a template of a cmake file"""
+    lines = (
+        f"project({project_name} LANGUAGES CXX)\n\n"
+        "cmake_minimum_required(VERSION 3.13 FATAL_ERROR)\n\n"
+        "set(CMAKE_CXX_STANDARD 17)\n"
```
Comment (jlhcrawford) [inline kit/command_new.py:34]: Weren't we going to move these values (17 and 3.13) to some "constant" variables?

File: kit/constants.py [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
+    """Defines constants for the he-toolkit"""
 
+    # version and the docker's tags
     user: str = getuser()
     version: str = "2.0.0"
     base_label: str = f"{user}/ubuntu_he_base:{version}"
     toolkit_label: str = f"{user}/ubuntu_he_toolkit:{version}"
     vscode_label: str = f"{user}/ubuntu_he_vscode:{version}"
+
+    # cmakes properties
```
Comment (jlhcrawford) [inline kit/constants.py:21]:
```suggestion
    # cmake properties
```

File: kit/README.md [hunk 3]
```diff
@@ -133,6 +133,40 @@ In order to uninstall a specific component, execute the remove command
 hekit remove hexl 1.2.3
 ```
 
+The command new can be used to create a new application. When it is executed
```
Comment (jlhcrawford) [inline kit/README.md:136]:
```suggestion
The command `new` can be used to create a new project. When it is executed
```

File: kit/README.md [hunk 4]
```diff
@@ -133,6 +133,40 @@ In order to uninstall a specific component, execute the remove command
...
+with the option `-–based-on`, it will create a copy of the base project,
+keeping its directory structure.
+```bash
+hekit new my-secure-query --based-on secure-query
+```
+
+However, when the command is executed without `-–based-on`.
```
Comment (jlhcrawford) [inline kit/README.md:143]:
```suggestion
However, when the command is executed without `-–based-on`
```

File: kit/README.md [hunk 5]
```diff
@@ -133,6 +133,40 @@ In order to uninstall a specific component, execute the remove command
...
+      |- README.md
+      └── include
+            └── example.h
+      └── recipes
+            └── example.toml
+      └── src
+            └── example.cpp
+```
+
+And the following actions should be completed to build the new application:
```
Comment (jlhcrawford) [inline kit/README.md:162]:
```suggestion
The following actions should be completed to build the new application:
```

File: kit/README.md [hunk 6]
```diff
@@ -133,6 +133,40 @@ In order to uninstall a specific component, execute the remove command
...
+
+* Open the toml file inside the recipes directory and replace `-DFLAG=TBD` with the required flags
+
+* Open CMakeLists.txt and uncomment the statements for find_package and target_link_libraries of the required library. If other dependencies are needed, for instance Threads, the file must be updated to include and compile them.
```
Comment (jlhcrawford) [inline kit/README.md:166]:
```suggestion
* Open `CMakeLists.txt` and uncomment the statements for `find_package` and `target_link_libraries` of the required library. If other dependencies are needed, for instance `Threads`, the file must be updated to include and compile them.
```

File: kit/README.md [hunk 7]
```diff
@@ -133,6 +133,40 @@ In order to uninstall a specific component, execute the remove command
...
+            └── example.h
+      └── recipes
+            └── example.toml
+      └── src
+            └── example.cpp
+```
+
+And the following actions should be completed to build the new application:
+
+* Open the toml file inside the recipes directory and replace `-DFLAG=TBD` with the required flags
```
Comment (jlhcrawford) [inline kit/README.md:164]:
```suggestion
* Open the toml file inside the recipes directory and replace `-DFLAG=TBD` with the desired CMake flags for your project.
```

File: kit/README.md [hunk 8]
```diff
@@ -133,6 +133,40 @@ In order to uninstall a specific component, execute the remove command
...
+
+* Open CMakeLists.txt and uncomment the statements for find_package and target_link_libraries of the required library. If other dependencies are needed, for instance Threads, the file must be updated to include and compile them.
+
+* Add the code of the new application in the .cpp and .h files created by the command
```
Comment (jlhcrawford) [inline kit/README.md:168]:
```suggestion
* Add and/or write the code of the new application in the `.cpp` and `.h` files created by the command.
```

**Review by jlhcrawford — CHANGES_REQUESTED** [review]
Some typo/documentation changes. Otherwise, code looks good.

File: docker/Dockerfile.toolkit [hunk 9]
```diff
@@ -33,6 +33,7 @@ WORKDIR $HOME/he-toolkit
 RUN ./hekit init --default-config && \
     ./hekit install recipes/default.toml && \
     ./hekit build recipes/examples.toml --recipe_arg "toolkit-path=${HOME}/he-toolkit" && \
-    ./hekit build recipes/sample-kernels.toml --recipe_arg "toolkit-path=${HOME}/he-toolkit"
+    ./hekit build recipes/sample-kernels.toml --recipe_arg "toolkit-path=${HOME}/he-toolkit" && \
+    . docker/runners.sh
```
Comment (hamishun) [inline docker/Dockerfile.toolkit:37]: If this does not work, better to revert the change

**Review by hamishun — CHANGES_REQUESTED** [review]

**Review by hamishun — APPROVED** [review]

**Review by jlhcrawford — APPROVED** [review]

**Pull request merged**